### PR TITLE
Remove protect_env, improve logging

### PR DESCRIPTION
**Pull request by samtstern**
### Description

**Behavior Change**: we used to take all possible routes to block the user from accessing production services.  After #1479  (released in `7.2.0`) that is not the case.  Therefore our erasure of the `GOOGLE_APPLICATION_CREDENTIALS` is no longer helping devs and may harm them by not allowing them to select the right credentials.

**Other Changes**
  * Improve the logging style for functions to match what happens when you deploy functions (name in brackets as part of the log label).
  * Fix a bug where functions ignored because an emulator is not running were being logged in an ugly way.
  * New big warning if `GOOGLE_APPLICATION_CREDENTIALS` is set.
	 
### Scenarios Tested

```bash
$ GOOGLE_APPLICATION_CREDENTIALS=$HOME/service-accounts/fir-oss/keyfile.json npx firebase emulators:start --only functions,firestore
i  Starting emulators: ["functions","firestore"]
✔  functions: Using node@8 from host.
✔  functions: Emulator started at http://localhost:5001
i  firestore: Emulator logging to firestore-debug.log
✔  firestore: Emulator started at http://localhost:8080
i  firestore: For testing set FIRESTORE_EMULATOR_HOST=localhost:8080
i  functions: Watching "/tmp/tmp.tkVh7jwr4l/functions" for Cloud Functions...
⚠  functions: Your GOOGLE_APPLICATION_CREDENTIALS environment variable points to /usr/local/google/home/samstern/service-accounts/fir-oss/keyfile.json. Non-emulated services will access production using these credentials. Be careful!
✔  functions[helloJSON]: http function initialized (http://localhost:5001/fir-dumpster/us-central1/helloJSON).
✔  functions[info]: http function initialized (http://localhost:5001/fir-dumpster/us-central1/info).
✔  functions[writeToFirestore]: http function initialized (http://localhost:5001/fir-dumpster/us-central1/writeToFirestore).
✔  functions[writeToDatabase]: http function initialized (http://localhost:5001/fir-dumpster/us-central1/writeToDatabase).
✔  functions[firestoreReaction]: firestore function initialized.
i  functions[databaseReaction]: function ignored because the database emulator does not exist or is not running.
i  functions[authFunction]: function ignored because the firebaseauth.googleapis.com emulator does not exist or is not running.
✔  All emulators started, it is now safe to connect.
```

### Sample Commands

N/A

### CODE REVIEW

File: src/emulator/functionsEmulatorRuntime.ts [hunk 1]
```diff
@@ -956,8 +946,14 @@ async function main(): Promise<void> {
   }
 
   InitializeEnvironmentalVariables(frb);
-  if (isFeatureEnabled(frb, "protect_env")) {
```
Comment (samtstern) [inline src/emulator/functionsEmulatorRuntime.ts]: This is the behavior change.

Comment (matheo) [reply src/emulator/functionsEmulatorRuntime.ts]: @samtstern looks nice! the console output is awesome!

File: src/emulator/functionsEmulator.ts [hunk 2]
```diff
@@ -566,16 +566,10 @@ You can probably fix this by running "npm install ${
   addRealtimeDatabaseTrigger(
     projectId: string,
     definition: EmulatedTriggerDefinition
-  ): Promise<any> {
```
Comment (abeisgoat) [inline src/emulator/functionsEmulator.ts]: Keep an eye out for any of these `<any>` which should actually be `<void>`. This fix is right, I just think there are other excessive `any`

File: src/utils.ts [hunk 3]
```diff
@@ -106,7 +106,7 @@ export function logWarning(message: string, type = "warn"): void {
  * Log an info statement with a gray bullet at the start of the line.
  */
 export function logLabeledWarning(label: string, message: string, type = "warn"): void {
-  logger[type](clc.yellow.bold(`${WARNING_CHAR} ${label}:`), message);
+  logger[type](clc.yellow.bold(`${WARNING_CHAR}  ${label}:`), message);
```
Comment (abeisgoat) [inline src/utils.ts]: two spaces... why?

Comment (samtstern) [reply src/utils.ts]: All the other ones have two spaces, so the warnings were misaligned.

**Pull request merged**